Outbox implementation for events logging mechanism

k0oda opened the pull request:
For detailed documentation you can visit:
- [English version](docs/solution_en.md)
- [Russian version](docs/solution_ru.md)

k0oda committed:
implement outbox for events logging
add .venv and .ruff_cache to gitignore
add celery beat as requirement
fix CLICKHOUSE_HOST used for CLICKHOUSE_PORT setting
implement unit tests for outbox

mcproger commented on src/core/event_log_client.py:
```diff
@@ -44,13 +46,32 @@ def init(cls) -> Generator['EventLogClient']:
+    ) -> None:
+        try:
+            converted_data = self._convert_data(data)
+            with transaction.atomic():
```
Кажется транзакция помещена не совсем в правильное место – она открывается независимо от транзакции бизнес-логики в юзкейсе и может случиться ситуация, когда бизнес-логика отработает не так как ожидается, а событие попадет в outbox и система перейдет в неконсистентное состояние.

mcproger commented on src/outbox/tasks.py:
```diff
@@ -0,0 +1,38 @@
+logger = structlog.get_logger(__name__)
+
+@shared_task(bind=True)
+def process_outbox_events(self) -> None:  # noqa: ANN001
```
Логику обработки лучше не хранить на "инфраструктурном" уровне, типа селери тасок. Это усложняет тестирование и переиспользование

mcproger commented on src/users/use_cases/create_user.py:
```diff
@@ -55,7 +55,7 @@ def _execute(self, request: CreateUserRequest) -> CreateUserResponse:
 
     def _log(self, user: User) -> None:
         with EventLogClient.init() as client:
```
Здесь будет каждый раз открываться коннект в CH, хотя мы больше не пишем в кликхаус напрямую. Лишние сетевые коннекты будут плохо влиять на стабильность и перформанс всего приложения

k0oda committed:
move outbox processing to separated service
move insert_to_outbox into OutboxProcessor
move transaction from insert_to_outbox to CreateUser._execute

k0oda commented:
Перенес транзакцию, теперь она оборачивает саму бизнес-логику, и в случае ошибки эвент не будет записан в outbox
Также была перенесена логика обработки outbox-а в отдельный сервис, а таска теперь является оберткой, вызывающей сервис
Перенес insert_to_outbox из EventLogClient в сервис OutboxProcessor, дабы избежать лишних коннектов с ClickHouse и оставить EventLogClient исключительно для взаимодействия с ClickHouse
